src/main/java/com/solace/samples/java/patterns/DirectReplierBlocking.java

```diff
@@ -0,0 +1,156 @@
+
+        //5. Build and start the Receiver object
+        RequestReplyMessageReceiver requestReplyMessageReceiver = messagingService.requestReply().
+                createRequestReplyMessageReceiverBuilder().build(TopicSubscription.of(TOPIC_PREFIX + "*/direct/>"));
+        requestReplyMessageReceiver.start();
+        //5-A. Setup an event handler for situations where the reply message could not be published.
+        requestReplyMessageReceiver.setReplyFailureListener(failedReceiveEvent -> System.out.println("### FAILED RECEIVE EVENT " + failedReceiveEvent));
+
+        //6. Create an OutboundMessageBuilder for building the outbound reply message
+        final OutboundMessageBuilder outboundMessageBuilder = messagingService.messageBuilder();
+
+        //7. Define the handler for the incoming message.
+        final RequestReplyMessageReceiver.RequestMessageHandler messageHandler = (inboundMessage, replier) -> {
+
+            msgRecvCounter++;
+
+            //7-A. Since Direct messages, check if there have been any lost any messages
+            checkForDiscardedMessages(inboundMessage);
+
+            //This SOP is just for demo purposes, ideally considering the slow nature of console I/O, any such action should be avoided in message processing
+            System.out.println("The inbound message payload is : " + inboundMessage.getPayloadAsString());
+            System.out.println("The inbound message APPLICATION_MESSAGE_ID is : " + inboundMessage.getProperty(SolaceProperties.MessageProperties.APPLICATION_MESSAGE_ID));
+            System.out.println("The inbound message CORRELATION_ID is : " + inboundMessage.getProperty(SolaceProperties.MessageProperties.CORRELATION_ID));
+
+            //7-B. Create the outbound message payload.
+            StringBuilder outboundMessageStringPayload = new StringBuilder().append("Replying to message received on topic: ").append(inboundMessage.getDestinationName())
+                    .append(" with correlation id :").append(inboundMessage.getProperty(SolaceProperties.MessageProperties.CORRELATION_ID));
+
+            //7-C. Create the outbound message with headers and payload.
+            final OutboundMessage outboundMessage = outboundMessageBuilder
+                    .withProperty(SolaceProperties.MessageProperties.APPLICATION_MESSAGE_ID, inboundMessage.getProperty(SolaceProperties.MessageProperties.APPLICATION_MESSAGE_ID))
+                    .withProperty(SolaceProperties.MessageProperties.CORRELATION_ID, inboundMessage.getProperty(SolaceProperties.MessageProperties.CORRELATION_ID))
+                    .build(outboundMessageStringPayload.toString());
+
+            //This SOP is just for demo purposes, ideally considering the slow nature of console I/O, any such action should be avoided in message processing
+            System.out.println("The outbound message is : " + outboundMessage.getPayloadAsString());
+
+            //7-D. Post the reply to the incoming message
```

Comment:
For your own knowledge take a look at how the utility method handles the message IDs, does it take the request's message ID and add it as the correlation ID in the response? or does it do something else? I believe there is also a reply flag that gets set to make it work with the classic APIs ;) 

**Consider extending the comment to explain how this works if you think it isn't too much info.

Reply:
So I checked it out and indeed it generates a sequential identifier for each request : **#REQ0**, **#REQ1** and so on for each message that is published by the requestor. 
This id is used in the **CorrelationId** header.
The sequential identifier is specific to each session or execution.